lib/MusicBrainz/Server/Data/Collection.pm

```diff
@@ -111,6 +113,77 @@ sub is_empty {
     return $non_empty ? 0 : 1;
 }
 
+sub merge {
+    my ($self, $new_id, $old_ids, $user_id, %opts) = @_;
+
+    my @ids = ($new_id, @$old_ids);
+
+    my @collections = values %{ $self->c->model('Collection')->get_by_ids(@ids) };
+    my @collection_owners = uniq map { $_->editor_id } @collections;
+    if (any { $_ != $user_id } @collection_owners) {
+        confess('Attempt to merge collections by a different user');
+    }
+
+    $self->c->model('CollectionType')->load(@collections);
+    my @entity_types = uniq map { $_->type->item_entity_type } @collections;
+
+    if (@entity_types > 1) {
+        confess('Attempt to merge collections of different entity types');
+    }
+
+    my $type = $entity_types[0];
+
+    # Remove duplicate entities: for all entities that exist in multiple
+    # collections, remove them from all but one.
```

Comment:
Although we don't really use `added`, `position`, or `comment` right now, I feel like there should be some sort of precedence when merging those...

 * Prefer the earliest value for `added`?
 * Prefer the position/comment of the target entry, or if unset then the earliest entry for which those are set (based on `added`)?

Reply:
Comments could be concatenated, like annotations are, and let the user clean up. But not sure about the others... I guess earliest added and target position makes sense. I guess we would need to take extra steps before this one to clean this up then...

Reply:
Waiting is fine with me. Want to add a `# TODO` here at least?